Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.cpp

```diff
@@ -127,11 +298,100 @@ namespace EMStudio
         m_renderOptions = RenderOptions::Load(&settings);
     }
 
-    const RenderOptions* AtomRenderPlugin::GetRenderOptions() const
+    void AtomRenderPlugin::SaveRenderOptions()
+    {
+        AZStd::string renderOptionsFilename(GetManager()->GetAppDataFolder());
+        renderOptionsFilename += "EMStudioRenderOptions.cfg";
+        QSettings settings(renderOptionsFilename.c_str(), QSettings::IniFormat, this);
+        m_renderOptions.Save(&settings);
+    }
+
+    RenderOptions* AtomRenderPlugin::GetRenderOptions()
     {
         return &m_renderOptions;
     }
 
+    void AtomRenderPlugin::Render([[maybe_unused]]EMotionFX::ActorRenderFlagBitset renderFlags)
+    {
+        if (!m_animViewportWidget)
+        {
+            return;
+        }
+
+        AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+        AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, m_animViewportWidget->GetViewportContext()->GetId());
+        AzFramework::DebugDisplayRequests* debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+
+        namespace AztfVi = AzToolsFramework::ViewportInteraction;
+        AztfVi::KeyboardModifiers keyboardModifiers;
+        AztfVi::EditorModifierKeyRequestBus::BroadcastResult(
+            keyboardModifiers, &AztfVi::EditorModifierKeyRequestBus::Events::QueryKeyboardModifiers);
+
+        debugDisplay->DepthTestOff();
+        m_manipulatorManager->DrawManipulators(
+            *debugDisplay, m_animViewportWidget->GetCameraState(),
+            BuildMouseInteractionInternal(
+                AztfVi::MouseButtons(AztfVi::TranslateMouseButtons(QGuiApplication::mouseButtons())), keyboardModifiers,
+                BuildMousePick(m_animViewportWidget->mapFromGlobal(QCursor::pos()))));
+        debugDisplay->DepthTestOn();
```

Comment:
suggestion: Some of this looks very familiar too 😛 Again might be fine to leave for now, but would be cool to extract some of the units of work and share them here and in `EditorLib`.